src/inmanta/agent/agent.py

```diff
@@ -988,7 +988,7 @@ class AgentInstance:
     _get_resource_timeout: float
     _get_resource_duration: float
 
-    def __init__(self, process: "Agent", name: str, uri: str) -> None:
+    def __init__(self, process: "Agent", name: str, uri: str, ensure_deploy_on_start: bool = False) -> None:
```

Comment:
I have a slight preference to making this argument here and for the other methods kw-only. I find it improves readability a lot.